### ISSUE TITLE
It would be nice to be able to configure this to run only on files that have been modified.

### ISSUE THREAD

**Issue by patflynn** [sherter/google-java-format-gradle-plugin#13]
Basically include a flag in the plugin config so that it only runs on files that have been changed since the last commit.

**Comment by sherter** [comment]
It does that by default (both last modified time stamps and file sizes are recorded to detect changes). Why would you want to turn that off? (and yes, it _does_ detect if `toolVersion` has changed and re-formats everything if necessary)

**Comment by patflynn** [comment]
@sherter I haven't actually played with this. I just read the docs.  I assumed the first time I ran it on my project that it would reformat the whole thing.  I'd like to support the incremental approach of only formatting the changes I've made in the current workspace.

**Comment by sherter** [comment]
_All_ results are cached. Even the results of a verification task are reused for later invocations of formatting or verification tasks. Currently, a simple text file in `$buildDir/google-java-format/0.5/fileStates.txt` is used to record `<file, timestamp, size, state>` tuples for every file that has been processed (state being one of [these](https://github.com/sherter/google-java-format-gradle-plugin/blob/master/src/main/java/com/github/sherter/googlejavaformatgradleplugin/FileState.java)).

I would consider a plugin that _doesn't_ do this rather useless. But you're right, I should probably mention this feature somewhere...

**Comment by sherter** [comment]
Oh, I think I misunderstood the whole time, sorry! No, such a flag does currently not exist...

**Comment by patflynn** [comment]
@sherter yep. That makes sense.

**Comment by sherter** [comment]
@patflynn I've just released v0.6, which allows you to filter the inputs on the command line with a comma separated list of include patterns:
```shell
./gradlew <taskname> -D<taskname>.include="*Foo.java,bar/Baz.java"
```

Also see [contrib/pre-commit](https://github.com/sherter/google-java-format-gradle-plugin/blob/master/contrib/pre-commit) for a more useful example.

**Comment by patflynn** [comment]
@sherter  Thanks!
